### PR TITLE
ANSI-703 Hostname Aliasing Support

### PR DESCRIPTION
**Pull request by nerdynick**
# Description

This PR replaces #338 with a rebased change set. 

Adds support for node/host aliasing via the common use of `ansible_host` as well as the support for an addition new variable of `hostname` for an individual node. The chain of defaulting is captured in the `filter.py` file. The basic chain is `hostname`->`ansible_host`->`inventory_hostname`

Aliasing allows a user the foundations needed to install multiple components on a single node or a single component multiple times with separate configs on the same node. 

Fixes # ANSI-703

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

AWS EC2 infrastructure, test env.

**Test Configuration**:

3 ZK node cluster with a mix of all 3 addressing
3 KFK node cluster with a mix of all 3 addressing

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.variables/defaults/main.yml [hunk 1]
```diff
@@ -184,6 +184,9 @@ ssl_key_password: ""
 ### Boolean stating certs and keys are already on hosts. Used with ssl_custom_certs: true.
 ssl_custom_certs_remote_src: false
 
+### Enable Hostname Aliasing for host addressing. This will enable logic, on an individual host basis, to look for the variable `hostname`, followed by the reserved variable `ansible_host` and then `inventory_hostname` to resolve the appropriate FQDN of a host to use within configuration properties.
+hostaliasing: false
```
Comment (domenicbove) [inline roles/confluent.variables/defaults/main.yml:188]: wdyt about?
```suggestion
hostname_aliasing_enabled: false
```

Great annotation, be sure to run `python doc.py` to get the docs rewritten

**Review by domenicbove — APPROVED** [review]
LGTM! We should remember to include a sample inventory with the connect PR

**Comment by nerdynick** [comment]
Just found 2 bugs/missed changes that I need to address to wrap this up

**Review by domenicbove — APPROVED** [review]
LGTM

**Pull request merged**